src/view/ChatView.java

```diff
@@ -34,6 +44,35 @@ public ChatView(ChatController chatController, ChatViewModel viewModel) {
     this.add(scrollPane);
     this.add(messagePanel);
 
+    send.addActionListener(
+      // This creates an anonymous subclass of ActionListener and instantiates it.
+      new ActionListener() {
+        public void actionPerformed(ActionEvent evt) {
+          if (evt.getSource().equals(send)) {
+            ChatState currentState = viewModel.getState();
+
+            Message newMessage = new Message(Instant.now(), currentState.message);
+            chatController.sendMessage(newMessage);
+          }
+        }
+      });
+      message.addKeyListener(
+          new KeyListener() {
+            @Override
+            public void keyTyped(KeyEvent e) {
+              ChatState currentState = viewModel.getState();
+              String text = message.getText() + e.getKeyChar();
+              currentState.message = text;
```

Comment:
Prefer a setter function here

Reply:
Fixed by adding getter/setter to ChatState